### ISSUE TITLE
Duplicate namespaces in resources/stubs

### ISSUE THREAD

**Issue by Faboslav** [cycle/annotated#38]
For example there is `namespace Cycle\Annotated\Annotation;` located in `resources/stubs` and also in `src/Annotation`. This can't be right, right? It is messing with psalm and other alike tools. Is there any solution for that?

Error generated by psalm looks like this:
```
ERROR: InvalidNamedArgument - app/Entity/Trait/CreatedAtTrait.php:12:17 - Parameter $type does not exist on function Cycle\Annotated\Annotation\Column::__construct (see https://psalm.dev/238)
        #[Column(type: 'datetime')]
```

It is probably using the first occurence in alphabet for that namespace?

**Comment by butschster** [comment]
Hi @Faboslav.

`resources/stubs`  is used only for IDE autocompletion and you can ignore it.

**Comment by Faboslav** [comment]
Oh, i am ignoring it all right, but PSALM and PHPStan doesnt know that.

**Comment by butschster** [comment]
> Oh, i am ignoring it all right, but PSALM and PHPStan doesnt know that.

You can ignore vendor directory via psalm [config](https://psalm.dev/docs/running_psalm/configuration/)

```xml
<?xml version="1.0"?>
<psalm>
    <projectFiles>
        <directory name="src" />
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
</psalm>
```

**Comment by Faboslav** [comment]
But i want to resolve that error which Is generated by PSALM caused by two identical namespaces in this package, which Is in general bad practice, right?

**Comment by roxblnfk** [comment]
It is possible to make annotations with named arguments like this: https://github.com/cycle/annotated/pull/34
But we don't have priority in this direction. We have some task about release ORM v2.

**Comment by Faboslav** [comment]
What, do we understand each other? Annotations are working great, but why do we need to have two identical namespaces? That Is highly unusual right? And static analytic can't deal with that.

**Comment by roxblnfk** [comment]
Psalm does not check fake attributes (stubs) because they are not added in composer.json psr-4 section.
Static analysis cannot handle the signature of real attributes because they don't use named arguments.

---

Psalm sees this signature

https://github.com/cycle/annotated/blob/57be2acd6e962fc11e8a5a2bffe74b523a2a96c6/src/Annotation/Column.php#L54

and says that your definition `#[Column(type: 'datetime')]` is uncompatible.
But it works because under hood the spiral/attributes package converts `type: 'datetime'` from metadata to `['type' => 'datetime']` and pass it in the attribute constructor.

---

There are two solutions:
1 Configure suppression InvalidNamedArgument for Cycle Attributes
2 Update attributes to using named arguments like this: #34

**Comment by Faboslav** [comment]
Okay, i get it now, thanks for your explanation and patience.

**Comment by roxblnfk** [comment]
No problem.

By the way. Instead of [full ignoring some project files](https://github.com/cycle/annotated/issues/38#issuecomment-987016294) you can configure suppression for the `InvalidNamedArgument` handler only for cycle atributes. `psalm.xml` for example:

```xml
<?xml version="1.0"?>
<psalm
    ...
>
    ...
    <issueHandlers>
        ...
        <InvalidNamedArgument>
            <errorLevel type="suppress">
                <directory name="vendor/cycle/annotated/src/Annotation" />
            </errorLevel>
        </InvalidNamedArgument>
    </issueHandlers>
</psalm>
```

**Comment by roxblnfk** [comment]
Fixed in annotated v3.0.0 (Cycle ORM v2.0)

**Comment by Faboslav** [comment]
Oh this is cool, thank you for your hard work!
